File: Online-judge/leetcode/longest_palindromic_substring.go
func longestPalindrome(s string) string {
	var length int = len(s)

	if length < 0 {
		return s
	}
	
	var longestPalindrome string = ""
	var longestPalindromeLenght int = 0

	for index := 0; index < length; index++ {

		left, right := expandString(s, length, index, index)
		_left, _right := expandString(s, length, index, index+1)
		mx := len(s[left:right])
		_mx := len(s[_left:_right])

		if mx > _mx && mx > longestPalindromeLenght {
			longestPalindromeLenght = mx
			longestPalindrome = s[left:right]
		} else {
			if _mx > longestPalindromeLenght {
				longestPalindromeLenght = _mx
				longestPalindrome = s[_left:_right]
			}
		}
	}
	return longestPalindrome

}

func expandString(word string, length int, left int, right int) (int, int) {
	for left >= 0 && right < length && word[left] == word[right] {
		left--
		right++
	}
	return left + 1, right
}

File: Online-judge/leetcode/longest_palindromic_substring.py
class Solution(object):
    def longestPalindrome(self, s):

        length = len(s)
        if length < 0:
            return s
        longestString=''
        longestStringLength = 0

        for index in range(0, length):

            left ,right = self.expandPalindrome(s, length, index, index)
            _left ,_right = self.expandPalindrome(s, length, index, index+1)
            mx = len(s[left:right])
            _mx = len(s[_left:_right])

            if mx > _mx:
                if mx > longestStringLength:
                    longestString = s[left:right]
                    longestStringLength = mx
            else:
                if _mx > longestStringLength:
                    longestString = s[_left:_right]
                    longestStringLength = _mx
        return longestString

    def expandPalindrome(self, word, length, left, right):
        _left, _right = left, right
        
        while _left >= 0 and _right < length and word[_left] == word[_right]:
            _left -= 1
            _right += 1
        return _left+1, _right
